### PR TITLE
[O2B-1079] Fetch run number that includes a specific substring

### PR DESCRIPTION
**Pull request by saifrashed**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [ ] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- New filter option (not yet implemented in the front end)

Notable changes for developers:
- Added field in filter DTO for runs
- Added a conditional bock that checks if there is a pattern to search on
- Find run numbers with given integer as substring using the following regexp: d*/integer/d*
- Created a new test in runs.test.js

### CODE REVIEW

**Review by graduta — DISMISSED** [review]
Hi @saifrashed ,

You should have a look to use and extend the existing field `runNumbers` instead of adding a new parameter. 
The filter should be enriched so that it uses the same parameter. Moreover, it should not make an extra query to find those.

**Comment by martinboulais** [comment]
I think that making LIKE search on all substring will be hard to understand for user. Because if you want runs 1 & 3 (typing `1, 3`), you will also have run 11, 10, 31, 21 and so on. 
I propose, for start, to apply like only if there is no `,`. And if there is, search using oneof.
In a next step, we can think of allowing user to put `*` or `%` to put the fuzzy find even in comma separated lists, such as `123*, 234*` for example. But, this would be another ticket and need to be discussed further.

**Comment by martinboulais** [comment]
I think that making LIKE search on all substring will be hard to understand for user. Because if you want runs 1 & 3 (typing `1, 3`), you will also have run 11, 10, 31, 21 and so on. 
I propose, for start, to apply like only if there is no `,`. And if there is, search using oneof.
In a next step, we can think of allowing user to put `*` or `%` to put the fuzzy find even in comma separated lists, such as `123*, 234*` for example. But, this would be another ticket and need to be discussed further.

File: lib/usecases/run/GetAllRunsUseCase.js [hunk 1]
```diff
@@ -65,9 +65,18 @@ class GetAllRunsUseCase {
...
-                filteringQueryBuilder.where('runNumber').oneOfSubstrings(runNumberList.map((runNumber) => runNumber.trim()));
+                // If there is a comma we use oneOf to search runNumbers
+                if (runNumbers.includes(',')) {
+                    filteringQueryBuilder.where('runNumber').oneOf(...runNumberList);
+                }
+
+                // If there is no comma we use LIKE to search runNumbers
+                if (!runNumbers.includes(',')) {
+                    filteringQueryBuilder.where('runNumber').oneOfSubstrings(runNumberList);
+                }
```
[the remark this answers is not in the document]
Comment (saifrashed) [reply lib/usecases/run/GetAllRunsUseCase.js]: I have implemented the improved solution, thank you for the feedback. The previous solution did not make use of the already splitted runNumberList.

File: test/public/runs/overview.test.js [hunk 2]
```diff
@@ -672,7 +672,7 @@ module.exports = () => {
     });
 
     it('should successfully filter on a list of run numbers and inform the user about it', async () => {
```
Comment (martinboulais) [inline test/public/runs/overview.test.js:674]: Please add tests here on the substring filter, and same for the API test (runs api)

**Review by martinboulais — CHANGES_REQUESTED** [review]

**Review by martinboulais — APPROVED** [review]

**Pull request merged**